lib/puppet/application/doc.rb

```diff
@@ -51,81 +51,50 @@ def preinit
   def help
     <<-'HELP'
 
-puppet-doc(8) -- Generate Puppet documentation and references
+puppet-doc(8) -- Generate Puppet references
 ========
 
 SYNOPSIS
 --------
 Generates a reference for all Puppet types. Largely meant for internal
-Puppet Labs use.
+Puppet Labs use. (Deprecated)
 
 
 USAGE
 -----
-puppet doc [-a|--all] [-h|--help] [-l|--list] [-o|--outputdir <rdoc-outputdir>]
-  [-m|--mode text|pdf|rdoc] [-r|--reference <reference-name>]
-  [--charset <charset>]
+puppet doc [-h|--help] [-l|--list]
+  [-r|--reference <reference-name>]
 
 
 DESCRIPTION
 -----------
-If mode is not 'rdoc', then this command generates a Markdown document
+This deprecated command generates a Markdown document to stdout
 describing all installed Puppet types or all allowable arguments to
 puppet executables. It is largely meant for internal use and is used to
 generate the reference document available on the Puppet Labs web site.
 
-In 'rdoc' mode, this command generates an html RDoc hierarchy describing
-the manifests that are in 'manifest' and 'modulepath' configuration
-directives. The generated documentation directory is doc by default but
-can be changed with the 'outputdir' option.
+For all other use cases this command has been superseded by the "puppet-strings"
```

Comment:
I'd go ahead and say "For Puppet module documentation (and all other use cases)..."

Module docs are the main thing regular people used to do with this command. We should refer to it by name!